src/Symfony/Component/PropertyInfo/Extractor/PhpDocExtractor.php

```diff
@@ -227,9 +233,31 @@ private function getDocBlockFromMethod(string $class, string $ucFirstProperty, i
         }
 
         try {
-            return [$this->docBlockFactory->create($reflectionMethod, $this->contextFactory->createFromReflector($reflectionMethod)), $prefix];
+            return [$this->docBlockFactory->create($reflectionMethod, $this->createFromReflector($reflectionMethod)), $prefix];
         } catch (\InvalidArgumentException $e) {
             return null;
         }
     }
+
+    /**
+     * Prevents a lot of redundant calls to ContextFactory::createForNamespace.
+     *
+     * @param \ReflectionClass|\ReflectionMethod $reflector
+     */
+    private function createFromReflector($reflector): Context
+    {
+        if ($reflector instanceOf \ReflectionMethod)) {
+            $reflector = $reflector->getDeclaringClass();
```

Comment:
Why not moving the `getDeclaringClass()` call out here and add a proper `ReflectionClass` type-hint to the method parameter?

Reply:
done